blockchain/types/block.go

```diff
@@ -63,21 +63,32 @@ type Header struct {
 	Governance []byte `json:"governanceData"            gencodec:"required"`
 	Vote       []byte `json:"voteData,omitempty"`
 
-	BaseFee *big.Int `json:"baseFeePerGas,omitempty"    rlp:"optional"`
+	// Added with Magma hardfork for KIP-71 dynamic fee.
+	BaseFee *big.Int `json:"baseFeePerGas,omitempty" rlp:"optional"`
+
+	// Added with Randao hardfork for KIP-114 RANDAO.
+	RandomReveal []byte `json:"randomReveal,omitempty" rlp:"optional"` // 96 byte BLS signature
+	MixHash      []byte `json:"mixHash,omitempty" rlp:"optional"`      // 32 byte RANDAO mix
```

Comment:
`common.hash`?

Reply:
I used `[]byte` to distinguish between 'field not defined' and 'zero hash'.
Same for the RandomReveal.

Reply:
`RandomReveal != nil` is the same format with `RandomReveal != (common.RandomReveal{})`. 
I know that you prefer the first case. However, we have a potential error case against mismatched type, and it brings length check in almost usage. Could you reconsider it again?

Reply:
It's not a syntax issue. `[96]byte` and `[]byte` behaves differently in RLP encoding.

Recall that RLP encodes an array. The `rlp:"optional"` field may not present in the encoding, making the array short.
If we decode a short array into Header struct, `[96]byte` will be filled with 96 zero bytes, whereas `[]byte` will be nil. Refer to the test code below.

I need to clearly distinguish 'field not exists' and 'zero value' in consensus code as [we did with BaseFee field](https://github.com/klaytn/klaytn/blob/dev/consensus/istanbul/backend/engine.go#L206).

<details>
<summary>Details</summary>

```go
func TestHello(t *testing.T) {
	root := common.HexToHash("0xce26cd14bfe586c11f9ebae0186a3e4744dfb6233e603d477cdebc69d9dc7953")
	_ = root

	type Reveal [96]byte

	type A struct {
		Root   common.Hash
		Reveal Reveal      `rlp:"optional"`
		Mix    common.Hash `rlp:"optional"`
	}

	type B struct {
		Root   common.Hash
		Reveal []byte `rlp:"optional"`
		Mix    []byte `rlp:"optional"`
	}

	enc := func(x interface{}) {
		r, _ := rlp.EncodeToBytes(x)
		fmt.Printf("%x\n", r)
	}
	decA := func(rs string) {
		var x A
		rlp.DecodeBytes(hexutil.MustDecode(rs), &x)
		fmt.Printf("A{%x %x %x}\n", x.Root, x.Reveal, x.Mix)
	}
	decB := func(rs string) {
		var x B
		rlp.DecodeBytes(hexutil.MustDecode(rs), &x)
		fmt.Printf("B{%x %x %x}\n", x.Root, x.Reveal, x.Mix)
	}
	_, _, _ = enc, decA, decB

	decA("0xe1a0ce26cd14bfe586c11f9ebae0186a3e4744dfb6233e603d477cdebc69d9dc7953")
	decA("0xf8a4a0ce26cd14bfe586c11f9ebae0186a3e4744dfb6233e603d477cdebc69d9dc7953b860000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000")

	decB("0xe1a0ce26cd14bfe586c11f9ebae0186a3e4744dfb6233e603d477cdebc69d9dc7953")
	decB("0xf8a4a0ce26cd14bfe586c11f9ebae0186a3e4744dfb6233e603d477cdebc69d9dc7953b860000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000")

	//enc(A{root, Reveal{}, common.Hash{}})
	//enc(B{root, nil, nil})
	//enc(B{root, make([]byte, 96), make([]byte, 32)})
}
```

</details>

Reply:
It makes sense. Got it, thank you.